src/app/style-panel/style-panel.component.ts

```diff
@@ -0,0 +1,122 @@
+import { Component, AfterViewInit, ViewChildren } from '@angular/core';
+import { NgForm } from '@angular/forms';
+import { Store } from '@ngrx/store';
+import { FormArray, FormControl, FormGroup, Validators } from '@angular/forms';
+import { Observable, Observer, Subscriber, Subscription } from 'rxjs';
+
+import { getFields, getGeneralStyle } from '../core/store/index';
+import { StyleServices } from '../shared/style.services';
+import { getStyle } from '../core/store/index';
+import { ChangeStyleAction, RemoveStyleAction } from '../core/store/styles-fields/styleFields.actions';
+import { ChangeStyleGeneralAction } from '../core/store/styles-general/general-style.actions';
+import { valueDefault } from '../shared/value.sheets';
+import { GeneralStyle } from '../core/interfaces';
+
+enum EAttribute {
+  Placeholder = 'placeholder',
+  Required = 'required',
+  SelectValue = 'selectvalue',
+  SelectName = 'selectname'
+}
+
+@Component({
+  selector: 'app-style-panel',
+  templateUrl: './style-panel.component.html',
+  styleUrls: ['./style-panel.component.scss'],
+})
+export class StylePanelComponent implements AfterViewInit {
+  panelOpenState: boolean;
+  fieldsIsRenered: string[];
+  constructor(private store: Store, private styleServices: StyleServices) { }
+  arrStyleEntreis: any[];
+  generalStyle: GeneralStyle;
+  subGenStyle: Subscription;
+  subStyle: Subscription;
+
+  formGeneralStyle: FormGroup;
+  submitFormGeneralStyle(): void {
+
+    this.store.dispatch(
+      new ChangeStyleGeneralAction(this.formGeneralStyle.value)
+    );
+
+  }
+
+  ngOnInit() {
+    this.subGenStyle = this.store.select(getGeneralStyle).subscribe((v) => {
+
+      this.generalStyle = v;
+    });
+
+    console.log(999, this.subGenStyle); /////////
+
+    const {
+      stylesGeneral: { backgroundColor, height, width, border },
+      stylesGeneralInner: { backgroundColor: backgroundColorInner },
+    } = this.generalStyle;
```

Comment:
this.generalStyle comes from async function, may cause undefined error
think how to refactor this flow 
```suggestion
    } = this.generalStyle;
```